Incompatibility with the new version of Flink

The compile process will be failed with the new version of Flink because the following issue.

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:53 min
[INFO] Finished at: 2022-11-25T22:00:09+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project shuffle-plugin: Compilation failure
[ERROR] /home/github/flink-remote-shuffle/shuffle-plugin/src/main/java/com/alibaba/flink/shuffle/plugin/transfer/RemoteShuffleInputGate.java:[731,38] incompatible types: java.net.InetSocketAddress cannot be converted to org.apache.flink.runtime.taskmanager.TaskManagerLocation
[ERROR]
[ERROR] -> [Help 1]
```

The failed reason is that FLINK ConnectionID construction method is changed because of [FLINK-29639](https://issues.apache.org/jira/browse/FLINK-29639).

--- a/shuffle-plugin/src/main/java/com/alibaba/flink/shuffle/plugin/transfer/RemoteShuffleInputGate.java
+++ b/shuffle-plugin/src/main/java/com/alibaba/flink/shuffle/plugin/transfer/RemoteShuffleInputGate.java
@@ -34,6 +34,7 @@ import org.apache.flink.runtime.checkpoint.CheckpointOptions;
 import org.apache.flink.runtime.checkpoint.channel.ChannelStateWriter;
 import org.apache.flink.runtime.checkpoint.channel.InputChannelInfo;
 import org.apache.flink.runtime.checkpoint.channel.ResultSubpartitionInfo;
+import org.apache.flink.runtime.clusterframework.types.ResourceID;
 import org.apache.flink.runtime.deployment.InputGateDeploymentDescriptor;
 import org.apache.flink.runtime.deployment.SubpartitionIndexRange;
 import org.apache.flink.runtime.event.AbstractEvent;
@@ -57,6 +58,7 @@ import org.apache.flink.runtime.io.network.partition.consumer.RemoteInputChannel
 import org.apache.flink.runtime.io.network.partition.consumer.SingleInputGate;
 import org.apache.flink.runtime.jobgraph.IntermediateDataSetID;
 import org.apache.flink.runtime.shuffle.ShuffleDescriptor;
+import org.apache.flink.runtime.taskmanager.TaskManagerLocation;
 import org.apache.flink.runtime.throughput.ThroughputCalculator;
 import org.apache.flink.util.CloseableIterator;
 import org.apache.flink.util.FlinkRuntimeException;
@@ -72,6 +74,7 @@ import org.slf4j.LoggerFactory;
 import javax.annotation.concurrent.GuardedBy;
 
 import java.io.IOException;
+import java.net.InetAddress;
 import java.net.InetSocketAddress;
 import java.util.ArrayList;
 import java.util.Collection;
@@ -728,7 +731,10 @@ public class RemoteShuffleInputGate extends IndexedInputGate {
                     channelIndex,
                     new ResultPartitionID(),
                     0,
-                    new ConnectionID(new InetSocketAddress("", 0), 0),
+                    new ConnectionID(
+                            new TaskManagerLocation(
+                                    ResourceID.generate(), InetAddress.getLoopbackAddress(), 1),
+                            0),
                     new LocalConnectionManager(),
                     0,
                     0,

